src/display/canvas.js

```diff
@@ -1004,6 +1004,10 @@ var CanvasGraphics = (function CanvasGraphicsClosure() {
       composeSMask(this.ctx, this.current.activeSMask, groupCtx);
       this.ctx.restore();
       copyCtxState(groupCtx, this.ctx);
+
+      if (this.baseTransform) {
+        this.ctx.setTransform.apply(this.ctx, this.baseTransform);
```

Comment:
Since I'm not too familiar with this part of the code-base, I might be misunderstanding it completely, but isn't the `baseTransform` equal to `mozCurrentTransform` *before* rendering the SMask in this case?
Because the issue with this particular PDF file seems to be that `mozCurrentTransform` has changed during rendering of the SMask, and by using the `baseTransform` we're setting the transform back to the (initial) correct value!?